docs/upgrade.md

```diff
@@ -3,5 +3,263 @@ title: Upgrade guide
 description: Guide to update between major Powertools for AWS Lambda (TypeScript) versions
 ---
 
-!!! warning
-    This guide is a work in progress. We'll update it as we get closer to the 2.0 release. If you have any questions, or want to follow the progress, please join the discussion on the [GitHub issue](https://github.com/aws-powertools/powertools-lambda-typescript/issues/1714).
\ No newline at end of file
+## Migrate from v1 to v2
+
+To make your transition to v2 as smooth as possible while still providing you with the most up-to-date features and patterns we have made targeted breaking changes to some of the utilities. This guide will help you migrate your existing codebase to v2.
+
+### Quick summary
+
+Below a list of the main changes introduced in v2, review the sections below for more details and code examples for the changes that apply to your codebase.
+
+| Area                                   | Change                                                                                                                                           | Code change required |
+| -------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------ | -------------------- |
+| **ES Modules support**                 | Added ES Modules support via dual CommonJS and ES Module bundling, this enables top-level `await` and tree-shaking.                              | -                    |
+| **Middy.js middleware imports**        | Updated import path for Middy.js middlewares to leverage subpath exports - i.e. `@aws-lambda-powertools/tracer/middleware`.                      | Yes                  |
```

Comment:
```suggestion
| **Middy.js **        | Updated import path for Middy.js middlewares to leverage subpath exports - i.e. `@aws-lambda-powertools/tracer/middleware`.                      | Yes                  |
```

Could you tell me a bit more on `leverage subpath exports`? Is this for tree shaking?

Reply:
Yes, that's the main customer facing benefit.

This [comment here](https://github.com/aws-powertools/powertools-lambda-typescript/discussions/1269#discussioncomment-4891876) has a longer explanation of why tree-shaking wasn't working in v1, but essentially with subpath exports customers import that code path only if they actually need it.

If they don't, they have smaller bundle size and less code being evaluated during init.

In some cases (not for Middy specifically though), customers could have better tree-shaking already but it was a bit cumbersome and risky. If they inspected the file structure of the package they would see for instance that the code they wanted to import was found at `lib/formatters/LogFormatter.js`. 

With this knowledge they could import something by changing their imports to `@aws-lambda-powertools/logger/lib/formatters/LogFormatter.js`. This however was more verbose and also more likely to break in case we changed our directory structure or filename.


Also, as a closing note, tree shaking is a feature that [works only with ESM when using bundlers](https://esbuild.github.io/api/#tree-shaking) like `esbuild` (which is default in both SAM & CDK):
> Note that esbuild's tree shaking implementation relies on the use of ECMAScript module import and export statements. It does not work with CommonJS modules. Many packages on npm include both formats and esbuild tries to pick the format that works with tree shaking by default.

Reply:
thanks a lot for the super explainer ;) Lemme wordsmith it to see if I can make this slightly clearer.